pkg/www/retext/src/views/Home.vue

```diff
@@ -1,19 +1,47 @@
 <template>
   <div class="home">
-    <DocumentDisplay />
-    <HelloWorld msg="Two component test" />
+    <DocumentDisplay :channel="channel"  />
+    <CategoryList :channel="channel" />
   </div>
 </template>
 
 <script>
 // @ is an alias to /src
-import HelloWorld from '@/components/HelloWorld.vue'
+import CategoryList from '@/components/CategoryList'
 import DocumentDisplay from "@/components/DocumentDisplay";
 export default {
   name: 'Home',
   components: {
-    HelloWorld,
+    CategoryList,
     DocumentDisplay
+  },
+  data: () => {
+    return {
+      channel: {
+        obj: false
+      }
+    }
+  },
+  mounted() {
+    // don't know what to call this thing but channel doesn't really fit...
+    let channel = this.channel;
+    // eslint-disable-next-line no-unused-vars
+    // todo: make the order that one() and two() are called not matter
+    channel.one = (threeCb) => {
+      channel.obj = {threeCb: threeCb};
+    }
+    // eslint-disable-next-line no-unused-vars
+    channel.two = (obj, cb) => {
+      // todo: require three() to return a promise and call cb on the completion of that promise?
+      let x = channel.obj;
+      if(!x) // drag-drop didn't land on a category
+        return;
+
+      channel.obj = false;
+
+      x.threeCb(obj, cb);
+      return x;
```

Comment:
A few notes. 

1) Let's move this to its own file and class. In order to keep this logic our of the Home.vue file (since its technically a vue, but also it'll be better organization.

2) I suppose I understand that one/two need to be called in order.
A traditional channel blocks on both read and send until things are well, read and sent. Since this is javascript we won't quite get there. Therefore, I do like your idea of using a promise to emulate this behavior. The promise will allow you to always safely write and always safely read. 

3) that may also be a bit complicated, so using another go pattern (returning an [object, bool] tuple may also work. 

4) the x.threeCb method is a bit worrisome. 

5) I never said this explicitly, but let's keep brackets around our if statements. Remember that this project has probably more code/logic/mechanics on the frontend than the backend. We also have the luxury of a compiler/minifier so we should try to make our frontend code better than the backend code.